crates/sui-json-rpc/src/error.rs

```diff
@@ -76,19 +80,89 @@ impl Error {
                 }
                 _ => RpcError::Call(CallError::Failed(sui_error.into())),
             },
-            Error::QuorumDriverError(err) => match err {
-                QuorumDriverError::NonRecoverableTransactionError { errors } => {
-                    let error_object =
-                        ErrorObject::owned(-32000, NON_RECOVERABLE_ERROR_MSG, Some(errors));
-                    RpcError::Call(CallError::Custom(error_object))
-                }
-                _ => RpcError::Call(CallError::Failed(err.into())),
-            },
+            Error::QuorumDriverError(err) => match_quorum_driver_error(err),
             _ => RpcError::Call(CallError::Failed(self.into())),
         }
     }
 }
 
+pub fn match_quorum_driver_error(err: QuorumDriverError) -> RpcError {
+    match err {
+        QuorumDriverError::InvalidUserSignature(err) => {
+            let inner_error_str = match err {
+                // Use inner UserInputError's Display instead of SuiError::UserInputError, which renders UserInputError in debug format
+                SuiError::UserInputError { error } => error.to_string(),
+                _ => err.to_string(),
+            };
+
+            let error_message = format!("Invalid user signature: {inner_error_str}");
+
+            let error_object = ErrorObject::owned(INVALID_PARAMS_CODE, error_message, None::<()>);
+            RpcError::Call(CallError::Custom(error_object))
+        }
+        QuorumDriverError::TimeoutBeforeFinality
+        | QuorumDriverError::FailedWithTransientErrorAfterMaximumAttempts { .. } => {
+            RpcError::Call(CallError::InvalidParams(err.into()))
```

Comment:
Probably I don't understand this code well, so I have a question here: `TimeoutBeforeFinality` and `FailedWithTransientErrorAfterMaximumAttempts` are `InvalidParams` kind of errors?

Reply:
Oh great catch. No I suppose these are better categorized as transient server errors